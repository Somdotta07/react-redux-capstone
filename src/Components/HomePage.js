import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { getData } from '../Redux/home/home';

const HomePage = () => {
  const dispatch = useDispatch();
  const countryLists = useSelector((state) => state.covidReducer);
  useEffect(() => {
    if (countryLists.length === 0) {
      dispatch(getData());
    }
  }, []);

  const covidCases = countryLists.map((cases) => (
    <div className="country-name" id="country-n" key={cases.code}>
      <div className="country-c">
        <Link className="country-d" to={`/${cases.code}`}>
          {cases.name}
        </Link>
      </div>
      <div className="recovered">
        <div>
          Recovered:&nbsp;
          {cases.latest_data.recovered}
        </div>
      </div>
    </div>
  ));

  const searchCountry = () => {
    let inputText = '';
    const userInput = document.getElementById('input-country');
    const filterCountry = userInput.value.toUpperCase();
    const countryDetails = document.getElementsByClassName('country-name');
    [...countryDetails].forEach((country) => {
      const temp = country;
      inputText = temp.textContent || temp.innerText;
      if (inputText.toUpperCase().indexOf(filterCountry) > -1) {
        temp.style.display = '';
      } else {
        temp.style.display = 'none';
      }
    });
  };

  return (
    <div className="main-c">
      <div className="country-heading">
        <h2 className="line-1 anim-typewriter">Active Corona Cases </h2>
      </div>

      <input className="search" type="text" id="input-country" placeholder="Country name.." onKeyDown={searchCountry} />

      <div className="countries-con">
        {covidCases}
      </div>
    </div>

  );
};

export default HomePage;
