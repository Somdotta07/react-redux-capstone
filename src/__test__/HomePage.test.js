import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../Redux/configureStore';
import HomePage from '../Components/HomePage';

describe('Render components', () => {
  it('render HomePage', () => {
    const tree = renderer
      .create(<Router><Provider store={store}><HomePage /></Provider></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
