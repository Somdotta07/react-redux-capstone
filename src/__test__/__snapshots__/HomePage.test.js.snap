// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components render HomePage 1`] = `
<div
  className="main-c"
>
  <div
    className="country-heading"
  >
    <h2
      className="line-1 anim-typewriter"
    >
      Active Corona Cases 
    </h2>
  </div>
  <input
    className="search"
    id="input-country"
    onKeyDown={[Function]}
    placeholder="Country name.."
    type="text"
  />
  <div
    className="countries-con"
  />
</div>
`;
